server/api/middleware.go

```diff
@@ -86,3 +88,57 @@ type clusterCtxKey struct{}
 func getCluster(r *http.Request) *cluster.RaftCluster {
 	return r.Context().Value(clusterCtxKey{}).(*cluster.RaftCluster)
 }
+
+type auditMiddleware struct {
+	svr *server.Server
+}
+
+func newAuditMiddleware(s *server.Server) negroni.Handler {
+	return &auditMiddleware{svr: s}
+}
+
+// ServeHTTP is used to implememt negroni.Handler for auditMiddleware
+func (s *auditMiddleware) ServeHTTP(w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+	if !s.svr.IsServiceMiddlewareEnabled() || !s.svr.IsAuditMiddlewareEnabled() {
```

Comment:
When `IsAuditMiddlewareEnabled`, I think `IsServiceMiddlewareEnabled` must be enabled.

Reply:
Yes, I will remove `IsServiceMiddlewareEnabled ` is #4537